File: FSharp/BasicAlgorithms/FindAverageValue.fs
module FindAverageValue

let ForIn (numArr: int array) =
    let mutable sum = 0
    for (item ) in numArr do
      sum <- sum + item
    let average = sum/numArr.Length
    printfn "~~~ Find Average Value In Array -- For In Loop -- Value: %d ~~~" average

let ForTo (numArr: int array) = 
    let mutable sum = 0
    for index = 0 to numArr.Length - 1 do 
      sum <- sum + numArr[index]
    let average = sum/numArr.Length
    printfn "~~~ Find Average Value In Array -- For To Loop -- Value: %d ~~~" average
    

let WhileTo (numArr: int array)  = 
  let mutable index = 0
  let mutable sum = 0
  while index < numArr.Length do
    sum <- sum + numArr[index]
    index <- index + 1
  let average = sum/numArr.Length
  printfn "~~~ Find Average Value In Array -- While To Loop -- Value: %d ~~~" average

File: C/BasicAlgorithms/FindAverageValue.c
#include "basic_algorithms.h"

// Calculate and return the average of array elements
double find_average_value(const int arr[], size_t length) {
    if (length == 0) {
        return 0.0;
    }
    double sum = 0.0;
    for (size_t i = 0; i < length; i++) {
        sum += arr[i];
    }
    return sum / (double)length;
}
